### PR TITLE
Quarks IoTF RangeSensor Sample 

### PR DESCRIPTION
**Pull request by Alex-Cook4**
I have put together a sensor application that uses PI4J libraries to control a Proximity sensor HC-SR04 and LED and work through Watson IoT. 

This sample is based off of the IotfSensor sample, but uses real sensors. Since it depends on Pi4J libraries, I have updated the build files so that this sample only builds if there is a PI4J_LIB environment variable present. 

The full outline of this sensor application on a Pi is (I have bolded the main differences to the IotfSensor sample): 

**RangeSensor** -> Quarks Analytics (only send to IoTF in case of **close proximity average**) -> Watson IoT -> IBM Streams IoTF Sample -> Send command back to IoTF -> to Quarks on Pi -> **Light up LED to say we have received the command** 

This is my first commit to Quarks, so please let me know if I'm doing anything wrong or if I should be pointed to some specific docs. Thanks! 

Also, thank you to @dlaboss and @ddebrunner for helping me figure out how to get the build working correctly.

### CODE REVIEW

File: build.xml [hunk 1]
```diff
@@ -293,14 +293,16 @@
...
+      excludepackagenames="quarks.connectors.runtime,quarks.connectors.*.runtime,quarks.console.*,quarks.topology.spi.*,quarks.graph.spi.*,quarks.samples.scenarios.iotf.range.sensor"
       >
-      <sourcepath>
-         <dirset dir="${basedir}">
-           <include name="*/*/src/main/java"/>
-           <exclude name="android/**"/>
-         </dirset>
-      </sourcepath>
+  	
+    <sourcepath>
```
Comment (dlaboss) [inline build.xml]: looks like indentation was inadvertently changed?  if so, please revert it.

File: samples/scenarios/src/main/java/quarks/samples/scenarios/iotf/range/sensor/IotfRangeSensor.java [hunk 2]
```diff
@@ -0,0 +1,192 @@
+package quarks.samples.scenarios.iotf.range.sensor;
```
Comment (dlaboss) [inline samples/scenarios/src/main/java/quarks/samples/scenarios/iotf/range/sensor/IotfRangeSensor.java]: missing Apache license header.

File: samples/scenarios/src/main/java/quarks/samples/scenarios/iotf/range/sensor/IotfRangeSensor.java [hunk 3]
```diff
@@ -0,0 +1,192 @@
...
+import quarks.connectors.iot.IotDevice;
+import quarks.connectors.iot.QoS;
+import quarks.connectors.iotf.IotfDevice;
+import quarks.function.Supplier;
+import quarks.providers.direct.DirectProvider;
+import quarks.providers.direct.DirectTopology;
+import quarks.topology.TStream;
+import quarks.topology.TWindow;
+
+public class IotfRangeSensor {
```
Comment (dlaboss) [inline samples/scenarios/src/main/java/quarks/samples/scenarios/iotf/range/sensor/IotfRangeSensor.java]: This is a new file so can the indenting be made consistent within it wrt use of spaces vs tabs?  Things aren't lining up.  Note, as recently noted in DEVELOPMENT.md, the desire is no hard tabs.

File: samples/scenarios/src/main/java/quarks/samples/scenarios/iotf/range/sensor/IotfRangeSensor.java (hunk 2 above)
Comment (Alex-Cook4) [inline samples/scenarios/src/main/java/quarks/samples/scenarios/iotf/range/sensor/IotfRangeSensor.java]: Thanks. Added the Apache license headers.

File: samples/scenarios/src/main/java/quarks/samples/scenarios/iotf/range/sensor/IotfRangeSensor.java (hunk 3 above)
Comment (Alex-Cook4) [inline samples/scenarios/src/main/java/quarks/samples/scenarios/iotf/range/sensor/IotfRangeSensor.java]: I have fixed all the tabs to be spaces. Thanks for pointing that out.

**Comment by Alex-Cook4** [comment]
Do the checks rerun if they fail? I just ran the same build command: 

ant -Dquarks.build.ci=true -lib /usr/share/java/junit4.jar -lib jacoco-0.7.5/lib clean all test release

And it successfully built on my machine. I think it may have started building and I added several commits mid build? Not sure if that would be a problem...

**Comment by dlaboss** [comment]
Nah, if the "build" fails it doesn't run again until another commit is added or you close/reopen the PR.
In this case one of the WebSocketClient connector tests failed.  This has been happening intermittently -  QUARKS-122.

**Pull request merged**